core/domain/image_services.py

```diff
@@ -0,0 +1,103 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service for managing operations on images."""
+
+from core.domain import fs_domain
+from core.platform import models
+import feconf
+
+gae_image_services = models.Registry.import_gae_image_services()
+
+
+def get_image_dimensions(filename, exp_id):
+    """Gets the dimensions of the image with the given filename
+
+    Args:
+        filename: str. The filename of the image in exploration.
+        exp_id: str. ID of the exploration.
+
+    Returns:
+        tuple. Returns height and width of the image.
+    """
+    # Because the filepath in DEV_MODE is 'filename' whereas in PROD_MODE
```

Comment:
This is weird actually. The file-system interface should be a general one. You shouldn't have to be too careful about the path you pass into it. The internals of the file-system interface should handle that.

So I wouldn't expect to see this here.

Reply:
No, that needs to be done, because for the images we have.
DEV ---   exp_id/assets/random.png
Whereas for
GCS  ---- exp_id/assets/image/random.png

So we need to make sure the path passed is according to the system.

Reply:
Yeah. I have created the issue.

Reply:
OK, thanks. Can you link to it here? (In the future please do that too, to close out these threads properly.)

Reply:
https://github.com/oppia/oppia/issues/5445